### ISSUE TITLE
calling unsubscribe in React not unsubscribe observer immediately

### ISSUE THREAD

**Issue by jcouyang** [cujojs/most#360]
i'm working on a lib [react-most](https://github.com/reactive-react/react-most) to make most works with React, a issue bug me recently, what i'm expecting was that when i call unsubscribe imperatively in `componentWillUnmount()`, it should immediately "unsubscribe" the observer
```js
componentWillUnmount(){
  this.subscriptions.unsubscribe()
}
```
but after the component unmount, observer still get called. this will cause a unmounted component call it's `setState`.

in [componentDidMount](https://github.com/reactive-react/react-most/blob/master/lib/react-most.js#L57), i simply merge all streams and subscribe to setState. [L59](https://github.com/reactive-react/react-most/blob/master/lib/react-most.js#L59) is currently my work around to this issue

in [Todomvc](https://github.com/reactive-react/react-most/blob/master/examples/todomvc) example, the issue will happen when
- delete for example todo item with id 0
- there will be 2 streams react on this change, `mainSection`, and `todoItem`
1. `mainSection` will `setState` to remove item 0 in todo array,
2.  React will call todoItem's `componentWillUnmount`(which will call unsubscribe) then unmount the component
3. todoItem 0 itself will also react on this event, even step 2 already call unmount and `subscriptions.unsubscribe`, observer(setState) still been called on this unmounted component.  what i expected is todoItem 0 should stop calling it's observer in sync when unsubscirbe is imperatively called.

**Comment by jcouyang** [comment]
and this is not a React only issue, a smallest example to reproduce that would be...

1. this works fine, 2 will not print out since it's after unsubscribe in the sequence.
```js
var subscription = most.from([()=>1, subscription.unsubscribe, ()=>2])
  .subscribe({next:x=>console.log(x())})
```

2. but this does not work
```js
var subscription = most.just(1)
  .map(x=>subscription.unsubscribe())
  .map(()=>2)
  .subscribe({next:y=>{console.log(y)}})
```
2 will be print out, `unsubscribe()` should be called before next transform `map(()=>2)`, but imperative unsubscribe seems not work on this case.

**Comment by briancavalier** [comment]
Closed by #361
